app/src/main/java/com/asfoundation/wallet/nfts/ui/nftdetails/NFTDetailsFragment.kt

```diff
@@ -0,0 +1,99 @@
+package com.asfoundation.wallet.nfts.ui.nftdetails
+
+import android.graphics.Color
+import android.graphics.drawable.Drawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import by.kirich1409.viewbindingdelegate.viewBinding
+import com.asf.wallet.R
+import com.asf.wallet.databinding.FragmentNftBinding
+import com.asfoundation.wallet.base.SingleStateFragment
+import com.asfoundation.wallet.viewmodel.BasePageViewFragment
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.engine.DiskCacheStrategy
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
+import com.bumptech.glide.request.RequestListener
+import javax.inject.Inject
+
+class NFTDetailsFragment : BasePageViewFragment(),
+    SingleStateFragment<NFTDetailsState, NFTDetailsSideEffect> {
+
+  @Inject
+  lateinit var viewModelFactory: NFTDetailsViewModelFactory
+
+  @Inject
+  lateinit var navigator: NFTDetailsNavigator
+
+  private val viewModel: NFTDetailsViewModel by viewModels { viewModelFactory }
+  private val views by viewBinding(FragmentNftBinding::bind)
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                            savedInstanceState: Bundle?): View? {
+    return inflater.inflate(R.layout.fragment_nft, container, false)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    viewModel.collectStateAndEvents(lifecycle, viewLifecycleOwner.lifecycleScope)
+    setListeners()
+  }
+
+  fun ImageView.load(url: String?, onLoadingFinished: () -> Unit = {}) {
+    val listener = object : RequestListener<Drawable> {
+      override fun onLoadFailed(
+          e: GlideException?,
+          model: Any?,
+          target: com.bumptech.glide.request.target.Target<Drawable>?,
+          isFirstResource: Boolean
+      ): Boolean {
+        onLoadingFinished()
+        this@load.setBackgroundColor(Color.parseColor("#000000"))
```

Comment:
This shouldn't use hardcoded values, you can use instead

`ContextCompat.getColor(this, R.color.black)` or `ResourcesCompat.getColor(resources, R.color.black, null)`